exchange/auction.go

```diff
@@ -179,9 +226,9 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 		}
 	}
 
-	if debugLog != nil && debugLog.Enabled {
-		debugLog.BuildCacheString()
+	if len(toCache) > 0 && debugLog != nil && debugLog.Enabled {
```

Comment:
So here we wanna add debug log if there is something to cache? Wouldn't `toCache` greater than 0 mean we had valid bids?

Reply:
Yes, so this check is here due to a case where the debug log caching is attempted, but no bids make it to the cache. For example, if there are bids returned but they are all rejected due to not having a category, this caching function would still be called. What was happening was that there was still a call to the cache for the debug log, but it didn't have a proper cache key. So if this check fails, we do the debug log later when we check if there were any adpods generated.

Reply:
So if all the bids were say rejected then basically the above for loop won't execute and which would mean that `toCache` would be of length `0`, no? The current check to me looks like there are bids to be cached but we're still attempting debug log caching which is ok if we wanna do debug log caching as long as debugging is enabled and no matter if there's an error or not.

Reply:
Yes, so this won't change debug log behavior based off of errors. It is only based off of if we are attempting to cache bids, mostly due to needing the use the same cache key as the bids.